src/middleware/events.coffee

```diff
@@ -90,66 +95,58 @@ storeVisualizerEvents = (ctx, done) ->
   # round a sub MIN ms response to MIN ms
   if endTS-startTS<minEvPeriod then endTS = startTS+minEvPeriod
 
-  # Transaction start for channal
-  trxEvents.push
-    ts: startTS
-    comp: "channel-#{ctx.authorisedChannel.name}"
-    ev: 'start'
-  # Transaction start for primary route
+  # Transaction end for primary route
   trxEvents.push
-    ts: startTS
-    comp: ctx.authorisedChannel.name
-    ev: 'start'
+    channelID: ctx.authorisedChannel._id
+    transactionID: ctx.transactionId
+    visualizerTimestamp: startTS
+    route: 'primary'
+    event: 'start'
+    name: ctx.authorisedChannel.name
 
   if ctx.routes
     # find TS difference
     tsDiff = getTsDiff startTS, ctx.routes
 
     for route in ctx.routes
-      addRouteEvents trxEvents, route, 'route', tsDiff
+      addRouteEvents ctx, trxEvents, route, 'route', tsDiff
 
       if route.orchestrations
         # find TS difference
         tsDiff = getTsDiff startTS, route.orchestrations
-        addRouteEvents trxEvents, orch, 'orch', tsDiff for orch in route.orchestrations
+        addRouteEvents ctx, trxEvents, orch, 'orchestration', tsDiff for orch in route.orchestrations
   if ctx.mediatorResponse?.orchestrations?
     # find TS difference
     tsDiff = getTsDiff startTS, ctx.mediatorResponse.orchestrations
-    addRouteEvents trxEvents, orch, 'orch', tsDiff for orch in ctx.mediatorResponse.orchestrations
-
-  status = 200
-  if ctx.transactionStatus is messageStore.transactionStatus.COMPLETED
-    status = 'completed'
-  else if ctx.transactionStatus is messageStore.transactionStatus.COMPLETED_W_ERR
-    status = 'completed-w-err'
-  else if ctx.transactionStatus is messageStore.transactionStatus.FAILED
+    addRouteEvents ctx, trxEvents, orch, 'orchestration', tsDiff for orch in ctx.mediatorResponse.orchestrations
+
+  status = 'success'
+  if 500 <= ctx.response.status <= 599
     status = 'error'
 
   # Transaction end for primary route
   trxEvents.push
-    ts: endTS + orchestrationTsBufferMillis
-    comp: ctx.authorisedChannel.name
-    ev: 'end'
-    status: status
-  # Transaction end for channel
-  trxEvents.push
-    ts: endTS + orchestrationTsBufferMillis
-    comp: "channel-#{ctx.authorisedChannel.name}"
-    ev: 'end'
-    status: status
+    channelID: ctx.authorisedChannel._id
+    transactionID: ctx.transactionId
+    visualizerTimestamp: endTS + orchestrationTsBufferMillis
+    route: 'primary'
+    event: 'end'
+    name: ctx.authorisedChannel.name
+    status: ctx.response.status
+    visualizerStatus: status
 
   now = new Date
   event.created = now for event in trxEvents
-  events.VisualizerEvent.collection.ensureIndex { created: 1 }, { expireAfterSeconds: 600 }, ->
-    events.VisualizerEvent.collection.insert trxEvents, (err) -> return if err then done err else done()
+  events.Event.collection.ensureIndex { created: 1 }, { expireAfterSeconds: 3600 }, ->
+    events.Event.collection.insert trxEvents, (err) -> return if err then done err else done()
```

Comment:
It's apparently much better in terms of performance to do it this way[*](http://stackoverflow.com/questions/16726330/mongoose-mongodb-batch-insert)[*](https://groups.google.com/forum/#!msg/mongoose-orm/IkPmvcd0kds/bZuYCN_orckJ). It looks like the only method that mongoose has for doing batch insert is `.create`, but it sounds like that basically just loops and calls `.save` for each document, rather than doing a proper batch insert.